src/components/ApiSearch/ApiSearch.tsx

```diff
@@ -34,98 +33,77 @@ export const ApiSearchFunc = forwardRef(
-              'data-[side=bottom]:translate-y-1 data-[side=left]:-translate-x-1 data-[side=right]:translate-x-1 data-[side=top]:-translate-y-1',
-              className
+        <Ariakit.ComboboxProvider>
+          <div className="flex flex-col">
+            {label && (
+              <label className={cn('text-muted-foreground', labelClassName)}>
+                {label}
+              </label>
+            )}
+            <Ariakit.Combobox
+              ref={ref}
+              className={cn(
+                className,
+                'combobox w-full',
+                'flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50',
+                error ? 'border-[1px] border-red-600' : ''
+              )}
+              {...rest}
+              onChange={(ev) => {
+                setSelectedItem({
+                  name: ev.target.value,
+                });
+                setSearch(ev.target.value);
+              }}
+            />
+            {helperText && (
+              <p className={'text-red-600 text-sm'}>{helperText}</p>
             )}
-          >
-            {filteredItems.length > 0 ? (
-              filteredItems.map((supply) => (
+          </div>
+          {filteredItems.length > 0 && (
+            <Ariakit.ComboboxPopover
+              gutter={4}
+              sameWidth
+              className={cn(
+                'popover w-full z-10 max-h-96 p-1',
+                'overflow-x-hidden	overflow-y-scroll rounded-md border bg-popover text-popover-foreground shadow-md data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2',
+                className
+              )}
+            >
+              {filteredItems.map((item) => (
                 <>
-                  <div
-                    className={cn(
-                      'p-1 hover:bg-accent hover:text-accent-foreground',
-                      'h-[var(--radix-select-trigger-height)] w-full min-w-[var(--radix-select-trigger-width)]',
-                      className
-                    )}
+                  <Ariakit.ComboboxItem
+                    key={item.id}
                     onClick={() => {
-                      setItem(supply.name);
                       setSelectedItem({
-                        name: supply.name,
-                        id: supply.id,
+                        name: item.name,
+                        id: item.id,
                       });
```

Comment:
Se item tiver apenas `id` e `name` poderia ser apenas:
```suggestion
                      setSelectedItem(item);
```

Reply:
booa!